src/Gui/ApplicationPy.cpp

```diff
@@ -786,13 +784,11 @@ PyObject* Application::sExport(PyObject * /*self*/, PyObject *args)
 
 
             SoGetPrimitiveCountAction action;
-            action.setCanApproximate(true);
+            action.setCanApproximate(1);

```

Comment:
Why change this ? The signature is a setCanApproximate(SbBool) so it clearly except a boolean (even if SbBool is an int, only 0 and 1 is allowed here, use C++'s true and false here)

Reply:
In the absence of any coding guidelines here it seemed a good idea to follow best practise warnings from clang-tidy. I can understand why that warning exists, but it also seems this is not a popular change here.

Reply:
I'm not a FreeCAD developer. So this is my opinion here, I'll let the official dev kicks in to correct. Anyway, this code relies on OpenInventor library which is a mixed C & C++ library. So they "redefined" bool type since it doesn't exist in C90. They named it `SbBool` which is quite clear about their intend. 

When you rely on a code analyzer, it doesn't care about the semantic and figure out that SbBool is a int, and you are feeding a boolean to it. The code analyzer here is wrong, because, clearly, only value 0 and 1 are expected here (it wouldn't choke if you had written `setCanApproximate(42)`. You don't know inside the function if the code is doing `if (param == TRUE)` or `if (param)`. The former would fail to work so you have to dig into the code to know that (and since it's a dependency, it's not in FreeCAD code base, so it's hard to find, you might not have it on your computer).

It's a perfect example of what I've written earlier, this kind of change increases the complexity for the developer for no actual benefit. 

Reading the new line of code triggers the query "is 1 some kind of threshold ? What is the unit ?"
while you should ask "ok, I'm allowing to make approximation".

Anyway, it's my 2 pence, FreeCAD developers will decide what they prefer.

Reply:
As already stated, I understand the argument for allowing implicit conversions. I know Werner supports their use and has said that maybe clang-tidy settings should be changed. But that hasn't happened yet.

Reply:
We can clean up clang-tidy later, let's drop this change from the PR.

Reply:
@wwmayer There are two available options for `readability-implicit-bool-conversion` [here](https://clang.llvm.org/extra/clang-tidy/checks/readability/implicit-bool-conversion.html) being: `
AllowIntegerConditions

    When true, the check will allow conditional integer conversions. Default is false.

AllowPointerConditions

    When true, the check will allow conditional pointer conversions. Default is false.
` Would false to the first, and true to the second be the way to go?